docs/topics/multiplatform/multiplatform-run-tests.md

```diff
@@ -1,67 +1,375 @@
+   `determineCurrentRuntime()` function:
+
+    ```kotlin
+    actual fun determineCurrentRuntime(): CurrentRuntime {
+        val name = System.getProperty("java.vm.name") ?: "Android"
+    
+        val version = System.getProperty("java.version")
+    
+        return CurrentRuntime(name, version)
+    }
+    ```
+
+3. Use the IDE's suggestions to create the `androidUnitTest/kotlin` directory:
+
+   ![Creating Android test directory](create-android-test-dir.png){width=300}
+
+4. In the `kotlin` folder, create a new `org.kmp.testing` directory.
+5. In this directory, create the `AndroidRuntimeTest.kt` file and update it with the following Android test:
+
+    ```kotlin
+    import kotlin.test.Test
+    import kotlin.test.assertContains
+    import kotlin.test.assertEquals
+    
+    class AndroidRuntimeTest {
+        @Test
+        fun shouldDetectAndroid() {
+            val runtime = determineCurrentRuntime()
+            assertContains(runtime.name, "OpenJDK")
+            assertEquals(runtime.version, "17.0")
+        }
+    }
+    ```
+
+It may seem strange that an Android-specific test is run on a local JVM. This is because these tests run as local unit
+tests on the current machine. As described in the [Android Studio documentation](https://developer.android.com/studio/test/test-in-android-studio),
+these tests are different from instrumented tests, which run on a device or an emulator.
+
+The **Kotlin Multiplatform App** template project is not configured to support these tests by default. However, it's
+possible to add additional dependencies and folders. To learn about adding support for instrumented tests, see this [Touchlab guide](https://touchlab.co/understanding-and-configuring-your-kmm-test-suite/)
+
+#### For iOS
+
+1. In the `iosMain/kotlin` folder, create a new `org.kmp.testing` directory.
+2. In this directory, create the `IOSRuntime.kt` file and update it with the actual implementation of the expected
+   `determineCurrentRuntime()` function:
+
+    ```kotlin
+    import kotlin.native.Platform
+    
+    actual fun determineCurrentRuntime(): CurrentRuntime {
+        val name = Platform.osFamily.name.lowercase()
+        return CurrentRuntime(name, null)
+    }
+    ```
+
+3. Use the IDE's suggestions to create the `iosTest/kotlin` directory:
+
+   ![Creating iOS test directory](create-ios-test-dir.png){width=300}
+
+4. In the `kotlin` folder, create a new `org.kmp.testing` directory.
+5. In this directory, create the `IOSRuntimeTest.kt` file and update it with the following iOS test:
+
+    ```kotlin
+    import kotlin.test.Test
+    import kotlin.test.assertEquals
+    
+    class IOSRuntimeTest {
+        @Test
+        fun shouldDetectOS() {
+            val runtime = determineCurrentRuntime()
+            assertEquals(runtime.name, "ios")
+            assertEquals(runtime.version, "unknown")
+        }
+    }
+    ```
+
+### Run multiple tests and read reports
```

Comment:
```suggestion
### Run multiple tests and generate reports
```

Reply:
The intended meaning here was to highlight that user should read reports
Maybe "analyze" would fit better here?

Reply:
Yes! It's just not clear what is "reading" otherwise.